lib/mapObjects/CGMarket.cpp

```diff
@@ -284,13 +285,13 @@ std::vector<int> CGBlackMarket::availableItemsIds(EMarketMode::EMarketMode mode)
 
 void CGBlackMarket::newTurn(CRandomGenerator & rand) const
 {
-	if(cb->getDate(Date::DAY_OF_MONTH) != 1) //new month
-		return;
-
-	SetAvailableArtifacts saa;
-	saa.id = id.getNum();
-	cb->pickAllowedArtsSet(saa.arts, rand);
-	cb->sendAndApply(&saa);
+	if(VLC->modh->settings.BLACK_MARKET_MONTHLY_ARTIFACTS_CHANGE && cb->getDate(Date::DAY_OF_MONTH) == 1) //new month and feature enabled
```

Comment:
Okay, though apart from this pull request - isn't it better practice to use "if" for main code logic and leave the rest for automatic return instead of explicitly quitting from function at beginning?